File: 2026-01/src/day_06/boj_25192/Main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);

    int n;
    cin >> n;

    set<string> st;
    int cnt = 0;

    while (n--) {
        string input;
        cin >> input;

        if (input == "ENTER") {
            cnt += st.size();
            st.clear();
        } else {
            st.insert(input);
        }
    }
    cnt += st.size();

    cout << cnt;
}

File: 2026-01/src/day_06/boj_25192/Main.java
package day_06.boj_25192;

import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int N = Integer.parseInt(br.readLine());
        Set<String> set = new HashSet<>();
        int cnt = 0;

        for (int i = 0; i < N; i++) {
            String input = br.readLine();

            if (input.equals("ENTER")) {
                cnt += set.size();
                set.clear();
            } else {
                set.add(input);
            }
        }
        cnt += set.size();

        System.out.println(cnt);
    }
}
